metricflow/test/model/validations/test_identifiers.py

```diff
@@ -196,3 +200,42 @@ def test_composite_identifiers_ref_and_name() -> None:  # noqa:D
                 materializations=[],
             )
         )
+
+
+def test_mismatched_identifier(simple_model__pre_transforms: UserConfiguredModel) -> None:  # noqa: D
+    """Testing two mismatched identifiers in two data sources
+
+    Add two identifiers with mismatched sub-identifiers to two data sources in the model
+    Ensure that our composite identifiers rule catches this incompatibility
+    """
+    model = copy.deepcopy(simple_model__pre_transforms)
+
+    bookings_source, _ = find_data_source_with(
+        model=model,
+        function=lambda data_source: data_source.name == "bookings_source",
+    )
+    listings_latest, _ = find_data_source_with(
+        model=model,
+        function=lambda data_source: data_source.name == "listings_latest",
+    )
+
+    identifier_bookings = Identifier(
+        name=IdentifierReference(element_name="composite_identifier"),
+        type=IdentifierType.FOREIGN,
+        identifiers=[CompositeSubIdentifier(ref="sub_identifier1")],
+    )
+    bookings_source.identifiers = flatten_nested_sequence([bookings_source.identifiers, [identifier_bookings]])
+
+    identifier_listings = Identifier(
+        name=IdentifierReference(element_name="composite_identifier"),
+        type=IdentifierType.FOREIGN,
+        identifiers=[CompositeSubIdentifier(ref="sub_identifier2")],
+    )
+    listings_latest.identifiers = flatten_nested_sequence([listings_latest.identifiers, [identifier_listings]])
```

Comment:
I know this is the current standard for these tests but I hate this copy-and-mutate approach. It makes the tests very hard to reason about. I guess we should leave it like this for the sake of consistency and just bulldoze it all whenever we take on #19